packages/features/audit-log/eventTypeAuditLogger.ts

```diff
@@ -0,0 +1,144 @@
+  IEventTypeLog,
+} from "./types/EventTypeAuditLogTypes";
+import { EventTypeAuditLogOption } from "./types/EventTypeAuditLogTypes";
+import deepDifference from "./util/deepDifference";
+
+abstract class EventTypeAuditLogger {
+  abstract readonly eventTypeAuditData: IEventTypeLog;
+  abstract readonly actionType: EventTypeAuditLogOption;
+
+  protected readonly actorUserId: number = 0;
+  protected readonly targetEventId: number = 0;
+  protected readonly targetTeamId: number = 0;
+  protected readonly targetEventType: EventType = {} as EventType;
+
+  constructor(actorUserId: number, targetEventType: EventType) {
+    if (!targetEventType.id || !targetEventType.teamId) return this;
+    this.actorUserId = actorUserId;
+    this.targetEventId = targetEventType.id;
+    this.targetTeamId = targetEventType.teamId;
+    this.targetEventType = targetEventType;
+  }
+
+  async log() {
+    if (this.eventTypeAuditData)
+      await prisma.auditLog.create({
+        data: this.eventTypeAuditData,
+      });
+  }
+}
+
+export class EventTypeCreateAuditLogger extends EventTypeAuditLogger {
+  actionType: typeof EventTypeAuditLogOption.EventTypeCreate;
+  eventTypeAuditData: IEventTypeCreateLog;
+
+  constructor(actorUserId: number, targetEventType: EventType) {
+    super(actorUserId, targetEventType);
+    this.actionType = EventTypeAuditLogOption.EventTypeCreate;
+    this.eventTypeAuditData = this.eventTypeCreateDataMaker(
+      this.actorUserId,
+      this.targetEventId,
+      this.targetTeamId
+    );
+  }
+
+  private eventTypeCreateDataMaker(
+    actorUserId: number,
+    targetEventId: number,
+    targetTeamId: number
+  ): IEventTypeCreateLog {
+    return {
+      actionType: this.actionType,
+      actorUserId,
+      target: {
+        targetEvent: targetEventId,
+      },
+      crud: CRUD.CREATE,
+      targetTeamId,
+    };
+  }
+}
+
+export class EventTypeUpdateAuditLogger extends EventTypeAuditLogger {
+  actionType: typeof EventTypeAuditLogOption.EventTypeUpdate;
+  eventTypeAuditData: IEventTypeUpdateLog;
+  private readonly requiredEventTypeChanged = ["title", "length", "locations", "slug", "schedulingType"];
+  private readonly prevEventType: EventType;
+
+  constructor(actorUserId: number, prevEventType: EventType, targetEventType: EventType) {
+    super(actorUserId, targetEventType);
+    this.prevEventType = prevEventType;
+
+    this.actionType = EventTypeAuditLogOption.EventTypeUpdate;
+    this.eventTypeAuditData = this.eventTypeUpdateDataMaker(
+      this.actorUserId,
+      this.targetEventId,
+      this.targetTeamId,
+      this.prevEventType,
+      this.targetEventType
+    );
+  }
+
+  private eventTypeUpdateDataMaker(
+    actorUserId: number,
+    targetEventId: number,
+    targetTeamId: number,
+    prevEventType: EventType,
+    targetEventType: EventType
+  ): IEventTypeUpdateLog {
+    const changedAttributes = deepDifference(prevEventType, targetEventType);
+    const changedAttributesList = [];
```

Comment:
One single record per update event instead of one record per changed field as discussed in our previous review.